### PR TITLE
Document Helm configuration of PDB and affinity/anti-affinity rules

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description

Document Helm configuration of PDB and affinity/anti-affinity rules for ExtAuth and RateLimit

# Context

Support for this functionality was added to Gloo Edge Enterprise with https://github.com/solo-io/solo-projects/pull/2946
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5648

### CODE REVIEW

File: docs/content/operations/production_deployment/_index.md [hunk 1]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
 
 You can scale up the `gateway-proxies` Envoy instances by using a Deployment or a DeamonSet. The amount of requests that the `gateway-proxies` can process increases with the amount of CPU that the `gateway-proxies` have access to.
 
+#### Pod Disruption Budget
+
+You can configure a Pod Disruption Budget (PDB) for the `gateway-proxy` deployment when installing Gloo Edge via Helm by setting the `gatewayProxies.NAME.PodDisruptionBudget` fields in your values override file.
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:188]:
```suggestion
To configure a Pod Disruption Budget (PDB) for the `gateway-proxy` deployment when you install Gloo Edge via Helm, set values for the `gatewayProxies.NAME.PodDisruptionBudget` fields in your values override file.
```

File: docs/content/operations/production_deployment/_index.md [hunk 2]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+
+For example, if the following snippet is used as an override when installing Gloo Edge via Helm...
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:190]:
```suggestion
For example, consider the following snippet of a values override files, which defines `minAvailable` as `2`.
```

File: docs/content/operations/production_deployment/_index.md [hunk 3]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+
+```yaml
+gatewayProxies:
+  gatewayProxy:
+    podDisruptionBudget:
+      minAvailable: 2
+```
+
+Then the the Pod Disruption Budget policy shown below will be created as a result:
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:199]:
```suggestion
As a result, the following pod disruption budget policy is created:
```

File: docs/content/operations/production_deployment/_index.md [hunk 4]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+  name: gateway-proxy-pdb
+  namespace: gloo-system
+spec:
+  minAvailable: 2
+  selector:
+    matchLabels:
+      gateway-proxy-id: gateway-proxy
+```
+
+[You can read more about pod disruption budgets in the kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets)
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:214]:
```suggestion
[You can read more about pod disruption budgets in the Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
```

File: docs/content/operations/production_deployment/_index.md [hunk 5]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+
+#### Affinity/Anti-Affinity
+
+You can configure affinity and anti-affinity rules on the `gateway-proxy` deployment by setting the `gatewayProxies.NAME.affinity` and `gatewayProxies.NAME.antiAffinity` fields, respectively, in your values override file.
+
+For example, the following snippet can be used to set affinity rules on the `gateway-proxy` deployment if used to override the Gloo Edge Helm installation:
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:220]:
```suggestion
For example, the following snippet of a values override file sets affinity rules on the `gateway-proxy` deployment:
```

File: docs/content/operations/production_deployment/_index.md [hunk 6]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+  selector:
+    matchLabels:
+      gateway-proxy-id: gateway-proxy
+```
+
+[You can read more about pod disruption budgets in the kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets)
+
+#### Affinity/Anti-Affinity
+
+You can configure affinity and anti-affinity rules on the `gateway-proxy` deployment by setting the `gatewayProxies.NAME.affinity` and `gatewayProxies.NAME.antiAffinity` fields, respectively, in your values override file.
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:218]:
```suggestion
To configure affinity and anti-affinity rules for the `gateway-proxy` deployment when you install Gloo Edge via Helm, set values for the `gatewayProxies.NAME.affinity` and `gatewayProxies.NAME.antiAffinity` fields, respectively, in your values override file.
```

File: docs/content/operations/production_deployment/_index.md [hunk 7]
```diff
@@ -183,6 +183,58 @@ Make sure you have the `%RESPONSE_FLAGS%` item in the log pattern.
...
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: kubernetes.io/e2e-az-name
+              operator: In
+              values:
+              - e2e-az1
+              - e2e-az2
+```
+
+[You can read more about affinity and anti-affinity in the kubernetes documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity).
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:237]:
```suggestion
[You can read more about affinity and anti-affinity in the Kubernetes documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity).
```

File: docs/content/operations/production_deployment/_index.md [hunk 8]
```diff
@@ -197,6 +249,35 @@ global:
         key: abcdef
 ```
 
+#### Pod Disruption Budgets and Affinity/Anti-Affinity rules
+
+Similar to the `gateway-proxy` deployment, a Pod Disruption Budget policy for the ExtAuth service can be established by setting `global.extensions.extAuth.deployment.podDisruptionBudget` in your Helm configuration file.
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:254]:
```suggestion
To configure a pod disruption budget for the ExtAuth service when you install Gloo Edge via Helm, set the `global.extensions.extAuth.deployment.podDisruptionBudget` field in your values override file.
```

File: docs/content/operations/production_deployment/_index.md [hunk 9]
```diff
@@ -197,6 +249,35 @@ global:
...
+
+By default, the ExtAuth service has the following `podAffinity` rule configured:
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:256]:
```suggestion
By default, the following `podAffinity` rule is configured for the ExtAuth service:
```

File: docs/content/operations/production_deployment/_index.md [hunk 10]
```diff
@@ -197,6 +249,35 @@ global:
...
+                matchLabels:
+                  gloo: gateway-proxy
+              topologyKey: kubernetes.io/hostname
+```
+
+ The default ExtAuth service affinity settings can be overwritten during installation by setting `global.extensions.extAuth.affinity` in your Helm configuration file. Additionally, anti-affinity rules for the ExtAuth service can be configured by setting `global.extensions.extAuth.antiAffinity`.
+
+# Rate Limit
+
+Similar to the `gateway-proxy` deployment, a Pod Disruption Budget policy for the `rate-limit` deployment can be established by setting `global.extensions.rateLimit.deployment.podDisruptionBudget` in your Helm configuration file.
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:277]:
```suggestion
To configure a pod disruption budget for the `rate-limit` deployment when you install Gloo Edge via Helm, set the `global.extensions.rateLimit.deployment.podDisruptionBudget` field in your values override file.
```

File: docs/content/operations/production_deployment/_index.md [hunk 11]
```diff
@@ -197,6 +249,35 @@ global:
...
+            podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  gloo: gateway-proxy
+              topologyKey: kubernetes.io/hostname
+```
+
+ The default ExtAuth service affinity settings can be overwritten during installation by setting `global.extensions.extAuth.affinity` in your Helm configuration file. Additionally, anti-affinity rules for the ExtAuth service can be configured by setting `global.extensions.extAuth.antiAffinity`.
+
+# Rate Limit
```
Comment (Rachael-Graham) [inline docs/content/operations/production_deployment/_index.md:275]: should this be ### to match ext-auth on 238? not sure where this fits on the page, i may be wrong

**Review by Rachael-Graham — COMMENTED** [review]
Docs review

File: docs/content/operations/production_deployment/_index.md (hunk 11 above)
Comment (ben-taussig-solo) [reply docs/content/operations/production_deployment/_index.md:275]: Good catch, yeah it should be!

**Comment by ben-taussig-solo** [comment]
Thanks for the review @Rachael-Graham! I've made the changes suggested in commit https://github.com/solo-io/gloo/pull/5789/commits/61eb290dbfb990e7f10300460a438143f3ad8b49 and am closing out your comments as a result

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**